### PR TITLE
[s]  Fixes the negative materials protolathe exploit

### LINKED ISSUE #26374: The protolathe can accept negative amounts of ores
image

### PR DESCRIPTION
**Pull request by BeeSting12**
:cl: BeeSting12
fix: The protolathe no longer accepts negative amounts of materials.
/:cl:

Fixes #26374

### CODE REVIEW

**Comment by BeeSting12** [comment]
shit

**Comment by Kromgar** [comment]
Some guy just inserted the negative materials and it could only be fixed by breaking open the lathe.


Also this was 4 hours after the orm negative exploit was supposed to be fixed. not sure if server was up to date or not

File: code/modules/research/protolathe.dm [hunk 1]
```diff
@@ -106,6 +105,8 @@ Note: Must be placed west/left of and R&D console to function.
 		if(!in_range(src, stack) || !user.Adjacent(src))
 			return
 		var/amount_inserted = materials.insert_stack(O,amount)
+		if (amount_inserted <= 0)
```
Comment (xxalpha) [inline code/modules/research/protolathe.dm]: Are you sure the problem is with the return value of insert_stack? It's not supposed to return negative values.

Comment (xxalpha) [inline code/modules/research/protolathe.dm]: The only way this proc can return a negative value is if the stack object being inserted has negative values.

Comment (BeeSting12) [inline code/modules/research/protolathe.dm]: I actually have no idea and this just looked like a thing that would work. I tested though and it seems to have fixed it.

**Comment by Tacolizard** [comment]
how long could this have existed?

File: code/modules/research/protolathe.dm (hunk 1 above)
Comment (xxalpha) [inline code/modules/research/protolathe.dm]: Do you know how to reproduce the bug?

Comment (BeeSting12) [inline code/modules/research/protolathe.dm]: Go to protolathe.
Get any type of resource that can be put in the protolathe.
Place it in the protolathe.
When it gives you the pop up, type in a negative number.
Check materials storage.

Comment (xxalpha) [inline code/modules/research/protolathe.dm]: I can't reproduce it with the steps you described, it fails silently if I try to insert a negative amount of sheets.

Comment (xxalpha) [inline code/modules/research/protolathe.dm]: As I said earlier, I can only reproduce this if I var edit the sheet stack to have a negative amount.

**Comment by octareenroon91** [comment]
Forever, if it's a side-effect bug of negative-amount stacks. ORM assumed too-sane input, lathes assumed sane stacks.

**Comment by xxalpha** [comment]
Negative amount stacks are not supposed to exist ever.

**Comment by Iamgoofball** [comment]
@xxalpha this is SS13, never assume something doesnt exist

VG had a bug where the clown's limbs would appear in the sub process list for the process scheduler a year back.

**Comment by xxalpha** [comment]
One of the coding guidelines here is to not sanity check everything because it might obscure bugs. This PR is one of those cases, the real bug isn't being fixed here, as far as I can tell, we're just patching a symptom.

**Comment by Iamgoofball** [comment]
What? Since when do we actively avoid sanity I get chewed out for that shit all the time

Requesting a maintainer

**Comment by RemieRichards** [comment]
There is a difference between having good sanity, and shit band aids.
If it doesn't fix the actual issue, it's a shit band aid.

**Comment by BeeSting12** [comment]
This just fixes protolathes accepting negative sheets. I don't know how negative sheets are acquired and it doesn't fix that

**Comment by xxalpha** [comment]
You are not supposed to need to fix that, that isn't the bug here.

**Comment by BeeSting12** [comment]
Closing because this isn't even the right way to fix it apparently

**Pull request closed without merging**